Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -530,10 +530,70 @@ std::optional<int64_t> parseUTCOffsetInMinutes(StringView string)
 template<typename CharacterType>
 static bool canBeCalendar(const StringParsingBuffer<CharacterType>& buffer)
 {
-    // https://tc39.es/proposal-temporal/#prod-Calendar
-    // Calendar :
-    //     [u-ca= CalendarName]
-    return buffer.lengthRemaining() >= 6 && buffer[0] == '[' && buffer[1] == 'u' && buffer[2] == '-' && buffer[3] == 'c' && buffer[4] == 'a' && buffer[5] == '=';
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // Step 4(a)(ii)(2)(a):
+    //  Let key be the source text matched by the AnnotationKey Parse Node contained within annotation
+
+    // https://tc39.es/proposal-temporal/#prod-Annotation
+    // Annotation :::
+    //     [ AnnotationCriticalFlag_opt AnnotationKey = AnnotationValue ]
+
+    // https://tc39.es/proposal-temporal/#prod-AnnotationCriticalFlag
+    // AnnotationCriticalFlag :::
+    //     !
+
+    // https://tc39.es/proposal-temporal/#prod-AnnotationKey
+    // AnnotationKey :::
+    //     AKeyLeadingChar
+    //     AnnotationKey AKeyChar
+
+    // https://tc39.es/proposal-temporal/#prod-AKeyLeadingChar
+    // AKeyLeadingChar :::
+    //     LowercaseAlpha
+    //     _
+
+    // https://tc39.es/proposal-temporal/#prod-AKeyChar
+    // AKeyChar :::
+    //     AKeyLeadingChar
+    //     DecimalDigit
+    //     -
```

Comment:
Please follow to the other comment's BNF to make it readable well. (See parseUTCOffsetInMinutes for example)
```
    // https://tc39.es/proposal-temporal/#prod-Annotation
    //
    // Annotation :::
    //     [ AnnotationCriticalFlag_opt AnnotationKey = AnnotationValue ]
    //
    // AnnotationCriticalFlag :::
    //     !
    //
    // AnnotationKey :::
    //     AKeyLeadingChar
    //     AnnotationKey AKeyChar
    //
    // AKeyLeadingChar :::
    //     LowercaseAlpha
    //     _
    //
    // AKeyChar :::
    //     AKeyLeadingChar
    //     DecimalDigit
    //     -
```

Also, I cannot see `AnnotationValue` definition.

Reply:
I think it's not necessary to include `AnnotationValue` in the comment because this code doesn't parse the value; it just looks ahead to see if a possible prefix of an `Annotation` is present (up to the `=`). What do you think? (The parsing is done by `parseOneCalendar()`. )

Reply:
Please use the above comment's content as is. The main purpose of this comment is understanding what is parsed in this function by checking this comment. So, we need necessary information (We do not want to check the spec to see whether this function is implemented correctly).

Reply:
Added `AnnotationValue` to comment.